crates/block-producer/src/state_view/mod.rs

```diff
@@ -163,23 +195,39 @@ fn ensure_in_flight_constraints(
         candidate_tx
             .input_notes()
             .iter()
-            .filter(|&commitment| already_consumed_nullifiers.contains(&commitment.nullifier()))
             .map(|commitment| commitment.nullifier())
+            .filter(|nullifier| already_consumed_nullifiers.contains(nullifier))
             .collect()
     };
 
     if !infracting_nullifiers.is_empty() {
         return Err(VerifyTxError::InputNotesAlreadyConsumed(infracting_nullifiers));
     }
 
+    // Check all notes not found in Store are in in-flight notes, return list of missing notes
+    let missing_notes: Vec<NoteId> = tx_notes_not_in_store
+        .iter()
+        .filter(|note_id| !notes_in_flight.contains(note_id))
+        .copied()
+        .collect();
+    if !missing_notes.is_empty() {
+        return Err(VerifyTxError::UnauthenticatedNotesNotFound(missing_notes));
+    }
+
     Ok(())
 }
 
+/// Ensures the constraints related to transaction inputs:
+/// - the candidate transaction's initial account state hash must be the same as the one
+///   in the Store or empty for new accounts
+/// - input notes must not be already consumed
+///
+/// Returns a list of input notes that were not found in the Store
```

Comment:
I would write the last sentence as: "Returns a list of unauthenticated input notes that were not found in the store."